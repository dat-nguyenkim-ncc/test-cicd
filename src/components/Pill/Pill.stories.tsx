import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import Pill, { PillProps } from './Pill'

export default {
  title: 'Primitives/Pill',
  component: Pill,
} as Meta

const Template: Story<PillProps> = args => <Pill {...args} />

export const Default = Template.bind({})
Default.args = {
  variant: 'primary',
} as PillProps
