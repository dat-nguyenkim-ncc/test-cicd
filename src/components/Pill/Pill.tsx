import React from 'react'
import { Box } from 'theme-ui'
import flags from 'country-code-emoji'
import { Paragraph } from '../primitives'
import { ViewInterface } from '../../types'

export type PillProps = ViewInterface<{
  variant?: keyof typeof variants
  label?: string
  flag?: string
  alt?: string | null
  icon?: string
}>

const variants = {
  primary: {
    bg: 'primary',
    color: 'white',
  },
  muted: {
    bg: 'gray04',
    color: 'text',
  },
  flag: {
    bg: 'white',
    color: 'text',
  },
  out: {
    bg: 'transparent',
    borderColor: 'primary',
    color: 'primary',
    borderWidth: 1,
    borderStyle: 'solid',
  },
}

export const mapIcon: Record<string, string> = {
  fintech: 'Fi',
  insurtech: 'In',
  regtech: 'Re',
  out: 'Out',
  Duplicated: 'Du',
}

const Pill = ({ alt, variant = 'primary', label, flag, icon, sx }: PillProps) => {
  let flagEmoji
  try {
    flagEmoji = flags(flag)
  } catch (e) {}

  return (
    <Box
      sx={{
        p: 2,
        maxHeight: '28px',
        minHeight: '28px',
        minWidth: '33px',
        textAlign: 'center',
        borderRadius: 6,
        ...variants[flagEmoji ? 'flag' : variant],
        ...sx,
      }}
    >
      {icon || label ? (
        <Paragraph bold={!!icon} sx={{ mt: '-1px', lineHeight: 1 }}>
          {icon ? mapIcon[icon] || '' : label}
        </Paragraph>
      ) : (
        <>
          {flag && flag.length === 2 && (
            <Box title={alt || ''} sx={{ mt: '-1px', cursor: 'help', maxWidth: '15px' }}>
              {flagEmoji}
            </Box>
          )}
        </>
      )}
    </Box>
  )
}

export default Pill
