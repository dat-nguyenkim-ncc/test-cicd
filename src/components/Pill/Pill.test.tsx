import * as React from 'react'
import { shallow } from 'enzyme'
import Pill from './Pill'

describe('Pill', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<Pill variant="primary" />)
    expect(wrapper).toMatchSnapshot()
  })
})
