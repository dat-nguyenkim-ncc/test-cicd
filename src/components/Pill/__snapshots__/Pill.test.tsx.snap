// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill matches snapshot 1`] = `ShallowWrapper {}`;
